includes/Core/Authentication/Clients/OAuth_Client.php

```diff
@@ -540,13 +540,19 @@ public function using_proxy() {
 	 * @return string URL to the setup page on the authentication proxy.
 	 */
 	public function get_proxy_setup_url( $access_code = '' ) {
-		$scope = rawurlencode( implode( ' ', $this->get_required_scopes() ) );
+		$scope        = rawurlencode( implode( ' ', $this->get_required_scopes() ) );
+		$redirect_url = $this->user_options->get( self::OPTION_REDIRECT_URL );
 
 		$query_params = array( 'scope' => $scope );
 		if ( ! empty( $access_code ) ) {
 			$query_params['code'] = $access_code;
 		}
 
+		if ( ! empty( $redirect_url ) ) {
+			$query_params['redirect'] = $redirect_url;
+			$this->user_options->delete( self::OPTION_REDIRECT_URL );
+		}
```

Comment:
@asvinb we don't need these changes in this method. The redirect will automatically be executed in the `OAuth_Client::authorize_user` method. So, just revert changes in this file and the redirect will work for you.

Reply:
Makes sense. Thanks!